src/lib/reasoners/intervals.mli

```diff
@@ -19,137 +19,140 @@
 
-val doesnt_contain_0 : t -> Th_util.answer
+module Int : EuclideanRing
+  with type explanation := Explanation.t
+   and type value := Z.t
+   and type 'a union = 'a union
+(** Union-of-intervals over integers. *)
 
-val is_positive : t -> Th_util.answer
+module Legacy : sig
+  (** The [Legacy] module reimplements (most of) the old legacy [Intervals]
+      module on top of the current implementation to ease the transition. *)
 
-val is_strict_smaller : t -> t -> bool
+  type t
 
-val new_borne_sup : Explanation.t -> Numbers.Q.t -> is_le : bool -> t -> t
+  exception NotConsistent of Explanation.t
+  exception No_finite_bound
 
-val new_borne_inf : Explanation.t -> Numbers.Q.t -> is_le : bool -> t -> t
+  val undefined : Ty.t -> t
 
-val only_borne_sup : t -> t
-(** Keep only the upper bound of the interval,
-    setting the lower bound to minus infty. *)
+  val point : Numbers.Q.t -> Ty.t -> Explanation.t -> t
 
-val only_borne_inf : t -> t
-(** Keep only the lower bound of the interval,
-    setting the upper bound to plus infty. *)
+  val doesnt_contain_0 : t -> Th_util.answer
 
-val is_point : t -> (Numbers.Q.t * Explanation.t) option
+  val is_strict_smaller : t -> t -> bool
 
-val intersect : t -> t -> t
+  val new_borne_sup : Explanation.t -> Numbers.Q.t -> is_le : bool -> t -> t
 
-val exclude : t -> t -> t
+  val new_borne_inf : Explanation.t -> Numbers.Q.t -> is_le : bool -> t -> t
 
-val mult : t -> t -> t
+  val only_borne_sup : t -> t
+  (** Keep only the upper bound of the interval,
+      setting the lower bound to minus infty. *)
 
-val power : int -> t -> t
+  val only_borne_inf : t -> t
+  (** Keep only the lower bound of the interval,
+      setting the upper bound to plus infty. *)
 
-val sqrt : t -> t
+  val is_point : t -> (Numbers.Q.t * Explanation.t) option
 
-val root : int -> t -> t
+  val intersect : t -> t -> t
 
-val add : t -> t -> t
+  val exclude : ?ex:Explanation.t -> Q.t -> t -> t
 
-val scale : Numbers.Q.t -> t -> t
+  val mult : t -> t -> t
 
-val affine_scale : const:Numbers.Q.t -> coef:Numbers.Q.t -> t -> t
-(** Perform an affine transformation on the given bounds.
-    Suposing input bounds (b1, b2), this will return
-    (const + coef * b1, const + coef * b2).
-    This function is useful to avoid the incorrect roundings that
-    can take place when scaling down an integer range. *)
+  val power : int -> t -> t
 
-val sub : t -> t -> t
+  val root : int -> t -> t
 
-val merge : t -> t -> t
+  val add : t -> t -> t
 
-val abs : t -> t
+  val scale : Numbers.Q.t -> t -> t
 
-val pretty_print : Format.formatter -> t -> unit
+  val affine_scale : const:Numbers.Q.t -> coef:Numbers.Q.t -> t -> t
+  (** Perform an affine transformation on the given bounds.
+      Supposing input bounds (b1, b2), this will return
+      (const + coef * b1, const + coef * b2).
+      This function is useful to avoid the incorrect roundings that
+      can take place when scaling down an integer range. *)
```

Comment:
I was confused for a moment because I did make that change. VScode was acting up and opening files in the wrong directory with about 50% chance yesterday, and it looks like I made this specific change in the wrong worktree, thanks for catching it!